exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/project/ProjectRecordBatch.java

```diff
@@ -193,6 +209,13 @@ protected IterOutcome doWork() {
             }
           }
           incomingRecordCount = incoming.getRecordCount();
+          memoryManager.update();
+          if (logger.isTraceEnabled()) {
+            logger.trace("doWork():[1] memMgr RC " + memoryManager.getOutputRowCount()
```

Comment:
use {} instead of concatenating strings with +

Reply:
Ok. Will do. But what is the advantage of using '{}' over static string concatenation with a check? I think format strings are more error(argument ordering errors) prone than concatenation.

Reply:
It's easier to read for the programmer. Also the string concatenation is only executed if the debug statement is executed. If you use + you have to have if statements around your debug statements to avoid any performance impact.

```
if (logger.isDebug()) {
 logger.debug("my " + variable + " message");
}
```

If you use the {} you don't have to have if statements in order to prevent a performance impact

```
logger.debug("my {} message", variable);
```

It's typically considered best practice to do the latter since it's much cleaner.